src/Swarm/Util/Yaml.hs

```diff
@@ -61,6 +62,9 @@ liftE = E . const
 withE :: Semigroup e => e -> With e f a -> With e f a
 withE e (E f) = E (f . (<> e))
 
+fromE :: (Monad f) => With e f e
```

Comment:
I would call this `getE`.  `fromE` sounds like you're converting `e` to some other type.

Reply:
I was considering `askE`, to make it look like `ReaderT`, but `getE` sounds good too.

Reply:
Yeah, `askE` is probably better!

Reply:
Too late, I used your suggestion :laughing:

Reply:
Hah! Either one is fine. 😄